docs/src/idioms/option-result.md

```diff
@@ -0,0 +1,12 @@
+# Options and Nullables
+
+In keeping with uniffi guidelines, we have made an effort to map core Rust concepts into their Typescript equivalents wherever possible, in order to allow the Typescript code to be as idiomatic and easy to work with as possible. A returned `Result<T, E>` from Rust will result in a flat type of `T` or a thrown `E`, while an `Option<T>` will become `T | undefined`. We believed that flattened types like this would be easier to work with than wrappers at every point for such common Rust primitives. This in in keeping with the approach taken by the core uniffi team for other languages (Kotlin turns `Option<T>` into `T`?).
+
+Note that this flattening does mean certain types which are perfectly legal (if ill advised) in Rust are not representable on the Uniffi layer. An `Option<Option<T>>` for example, could be represented in Rust, though we would assert it may be a poor stylistic choice even there. If you truly need to represent a tri-state, a wrapping enum with data feels like a clearer choice, and one that has first class support. We also note that the core uniffi project shares these limitations in some of its bindings (Kotlin), and it has not proven overly burdensome to date.
```

Comment:
```suggestion
Note that this flattening does mean certain types which are perfectly legal (if ill advised) in Rust are not representable on the Uniffi layer. An `Option<Option<T>>` for example, could be represented in Rust, though we would assert it may be a poor stylistic choice even there. If you need to represent a tri-state, an enum with three variants feels like a clearer choice, and one that has first class support. We also note that the core uniffi project shares these limitations in some of its bindings (Kotlin), and it has not proven overly burdensome to date.
```